### PR TITLE
Phase 3 - create GET a list of posts API for posts with pagination.

### PR DESCRIPTION
**Pull request by kanghowoo**
Phase 3
Let’s create GET a list of posts API for posts with pagination
We will use offset based pagination and we will return 10 posts in each page.

3-1 Mock implementation
Let’s create controller and service/storage interface for GET list of posts
And for service/storage implementations, please create mock implementation first.
Check your application API for GET working well and create PR

3-2 Use MySQL
Let’s implement storage implementation using MySQL.
Check your application API for GET list of posts working well with MySQL and create PR

### CODE REVIEW

File: src/main/java/com/mide/gangsaeng/board/BoardController.java [hunk 1]
```diff
@@ -16,6 +19,8 @@
 @RequestMapping("/boards")
 public class BoardController {
 
+    public static final String DEFAULT_PAGE = "0";
```
Comment (TK-one) [inline src/main/java/com/mide/gangsaeng/board/BoardController.java:22]: Why the page starts with 0?
Its confusing to client. Let's start with 1

And why it's type is string?

File: src/main/java/com/mide/gangsaeng/board/BoardServiceImpl.java [hunk 2]
```diff
@@ -29,4 +31,9 @@ public BoardResponse read(long id) {
                 .build();
     }
 
+    @Override
+    public List<Board> getPage(int offset, int size) {
```
Comment (TK-one) [inline src/main/java/com/mide/gangsaeng/board/BoardServiceImpl.java:35]: let's limit the max value of size.
Overwrite size to 30 if size is bigger than 30 to protect our resources.

**Review by TK-one — COMMENTED** [review]
Your service should show posts in the recent posts in first page. Please fix this.

File: src/main/java/com/mide/gangsaeng/board/BoardController.java (hunk 1 above)
Comment (kanghowoo) [reply src/main/java/com/mide/gangsaeng/board/BoardController.java]: I edited starts with 1. 
The reason its type is string is that the type of the `defaultValue` in `@RequestParam`.

Comment (TK-one) [reply src/main/java/com/mide/gangsaeng/board/BoardController.java]: okay. it's fuckin spring annotation.

**Review by TK-one — APPROVED** [review]

File: src/main/java/com/mide/gangsaeng/board/BoardController.java (hunk 1 above)
Comment (kanghowoo) [reply src/main/java/com/mide/gangsaeng/board/BoardController.java]: > okay. it's fuckin spring annotation.

fuckin....?

**Pull request merged**